File: gaussian_unknown_precision_jl/gaussian_unknown_precision_model.jl
import Pkg; Pkg.add("Distributions")
using Random, Distributions

function my_rnorm(n)
  d = Normal()
  rand(d, n)
end

function my_dlnorm(tau,m,s,l)
  μ_for_mean(m1, σ) = log(m1) - σ^2/2
  d = LogNormal(μ_for_mean(m, s), s)
  if (l==true)
    logpdf(d,tau)
  else
    pdf(d,tau)
  end
end

function my_rlnorm(n,m,s)
  μ_for_mean(m1, σ) = log(m1) - σ^2/2
  d = LogNormal(μ_for_mean(m, s), s)
  rand(d, n)
end

function my_dnorm(y,m,s,l)
  d = Normal(m,s)
  if (l==true)
    logpdf(d,y)
  else
    pdf(d,y)
  end
end

File: gaussian_unknown_precision_py/gaussian_unknown_precision_model.py
import numpy as np
from scipy.stats import lognorm
from scipy.stats import norm

def my_rnorm(n):
  return norm.rvs(size=n)

def my_dlnorm(tau,m,s,l):
  if (l==True):
    return lognorm.logpdf(tau,m,s)
  else:
    return lognorm.pdf(tau,m,s)

def my_rlnorm(n,m,s):
  return lognorm.rvs(s=s, loc=m, scale=1, size=n)

def my_dnorm(y,m,s,l):
  if (l==True):
    return norm.logpdf(y,m,s)
  else:
    return norm.pdf(y,m,s)
